### ISSUE TITLE
Error:convolution has unknown dim 

### ISSUE THREAD

**Issue by lutianfei** [onnx/tensorflow-onnx#312]
Hello I have a yolov3 model ,when I try to convert it onnx model,here is some error as follow:

> using tensorflow=1.5.0, onnx=1.4.1, opset=9, tfonnx=0.3.2/aa1141
2019-02-19 06:32:10.206243: I tensorflow/tools/graph_transforms/transform_graph.cc:264] Applying fold_constants
2019-02-19 06:32:10.261864: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2019-02-19 06:32:20.105056: I tensorflow/tools/graph_transforms/transform_graph.cc:264] Applying remove_attribute
2019-02-19 06:32:20.109477: I tensorflow/tools/graph_transforms/transform_graph.cc:264] Applying fold_batch_norms
2019-02-19 06:32:20.121146: I tensorflow/tools/graph_transforms/transform_graph.cc:264] Applying fold_old_batch_norms
2019-02-19 06:32:29.408945: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1105] Found device 0 with properties:
name: Tesla M40 major: 5 minor: 2 memoryClockRate(GHz): 1.112
pciBusID: 0000:04:00.0
totalMemory: 11.17GiB freeMemory: 11.06GiB
2019-02-19 06:32:29.408994: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1195] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: Tesla M40, pci bus id: 0000:04:00.0, compute capability: 5.2)
ERROR:tf2onnx:node conv2d_1/convolution has unknown dim [-1, 3, -1, -1] for pads calculation
ERROR:tf2onnx:node conv2d_1/convolution has unknown dim [-1, 3, -1, -1] for pads calculation
...
ERROR:tf2onnx:node conv2d_57/convolution has unknown dim [-1, 1024, -1, -1] for pads calculation
ERROR:tf2onnx:node conv2d_58/convolution has unknown dim [-1, 512, -1, -1] for pads calculation
ERROR:tf2onnx:node conv2d_58/convolution has unknown dim [-1, 512, -1, -1] for pads calculation
ERROR:tf2onnx:node conv2d_59/convolution has unknown dim [-1, 1024, -1, -1] for pads calculation
ERROR:tf2onnx:node conv2d_59/convolution has unknown dim [-1, 1024, -1, -1] for pads calculation
tensorflow ops: Counter({'Const': 205, 'Conv2D': 59, 'Mul': 58, 'Maximum': 58, 'BiasAdd': 51, 'Add': 23, 'Identity': 8, 'FusedBatchNorm': 8, 'Pad': 5, 'Placeholder': 1})
tensorflow attr: Counter({'T': 270, 'dtype': 206, 'value': 205, 'data_format': 118, 'dilations': 59, 'strides': 59, 'use_cudnn_on_gpu': 59, 'padding': 59, 'epsilon': 8, 'is_training': 8, 'Tpaddings': 5, 'shape': 1})
onnx mapped: Counter({'Const': 205, 'Conv2D': 59, 'Mul': 58, 'Maximum': 58, 'BiasAdd': 51, 'Add': 23, 'Identity': 8, 'FusedBatchNorm': 8, 'Pad': 5, 'Placeholder': 1, 'Transpose': 1})
onnx unmapped: Counter()
INFO:tf2onnx.optimizer.transpose_optimizer: 175 transpose op(s) left, ops diff after transpose optimization: Counter({'Transpose': 40, 'Conv': 0, 'Mul': 0, 'Max': 0, 'Pad': 0, 'BatchNormalization': 0, 'Add': -50})
WARNING:tf2onnx.optimizer.transpose_optimizer:please try add --fold_const to help remove more transpose

----------------------

the command I used refer to the tutorial:

`python -m tf2onnx.convert\
    --input yolo_0119_final.pb\
    --inputs input_1:0\
    --inputs-as-nchw input_1:0\
    --outputs conv2d_59/BiasAdd:0\
    --output yolo_0119_final.onnx\
    --verbose\
    --fold_const\
    --opset 9`

and whether I'm using --inputs-as-nchw param,and --opset or not , the error is always exists.

**Comment by nbcsm** [comment]
thanks for reporting.

for conv op, the image shape is used to calculate ```pads```, so the image shape need to be a const.
dynamic shape is supported (via ```auto_pad```) recently but not released to pypi yet.

could you give a try with latest master branch and let us know how it works please?

**Comment by nbcsm** [comment]
auto_pad is supposed to work, close this issue for now.
please re-open it if there is any issue.
